### PR TITLE
[INV-23] Add Docker & Docker compose for BE

### PR DESCRIPTION
**Pull request by Savesh1508**

### CODE REVIEW

**Comment by Savesh1508** [comment]
Screenshot_1
Screenshot_2
Screenshot_3

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      JWT_REFRESH_TOKEN_EXPIRY: ${JWT_REFRESH_TOKEN_EXPIRY}
+      COOKIES_REFRESH_MS: ${COOKIES_REFRESH_MS}
+    depends_on:
+      - db
+    networks:
+      - app-network
+
+  db:
+    container_name: db
+    image: postgres:14
```
Comment (NegatiV-64) [inline docker-compose.yml:31]: 14? Why not 17 or 16 at least?

File: tsconfig.json [hunk 2]
```diff
@@ -20,6 +20,9 @@
     "strictBindCallApply": false,
     "forceConsistentCasingInFileNames": false,
     "noFallthroughCasesInSwitch": false
-  }
+  },
+  "include": [
```
Comment (NegatiV-64) [inline tsconfig.json:24]: Why it was added? No need to modify the tsconfig

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
+FROM node:20-alpine
```
Comment (NegatiV-64) [inline Dockerfile:1]: Use 22 lts

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+RUN npm install
+
+COPY . .
+
+RUN npm run build
+
+EXPOSE 3000
+
+CMD ["npm", "run", "start:prod"]
```
Comment (NegatiV-64) [inline Dockerfile:15]: Devide the build into steps please

**Review by NegatiV-64 — CHANGES_REQUESTED** [review]

File: package.json [hunk 5]
```diff
@@ -10,7 +10,7 @@
     "start": "nest start",
     "start:dev": "nest start --watch",
     "start:debug": "nest start --debug --watch",
-    "start:prod": "node dist/main",
+    "start:prod": "node dist/src/main",
```
Comment (NegatiV-64) [inline package.json:13]: Don't change it

Comment (NegatiV-64) [reply package.json:13]: Make sure that Docker correctly copies the folder and files

**Review by NegatiV-64 — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,44 @@
+# STAGE 1: Installing dependencies
+FROM node:22 AS dependencies
+
+WORKDIR /app
+
+ENV APP_ENV=production
```
Comment (NegatiV-64) [inline Dockerfile:6]: I meant not APP ENV but NODE_ENV

Comment (NegatiV-64) [reply Dockerfile:6]: Please see comment above

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+WORKDIR /app
+
+COPY --from=dependencies /app/node_modules ./node_modules
+
+COPY . .
+
+RUN npm run build
+RUN npx tsc typeorm.config.ts --outDir dist
+RUN npx tsc database/migrations/*.ts --outDir dist/database/migrations
```
Comment (NegatiV-64) [inline Dockerfile:29]: Why do you need to compile them?

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,54 @@
+version: '3.9'
+
+services:
+  invention:
+    container_name: invention
+    image: invention/airbnb:1.0.0
+    build: .
+    ports:
+      - '${APP_PORT}:${APP_PORT}'
```
Comment (NegatiV-64) [inline docker-compose.yml:9]: I think that in EXPOSE in Dockerfile, you are writing 3000, why here it depends on the APP PORT?

File: docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+      COOKIES_REFRESH_MS: ${COOKIES_REFRESH_MS}
+      CORS_ENABLED: ${COOKIES_REFRESH_MS}
+      CORS_ALLOWED_ORIGINS: ${CORS_ALLOWED_ORIGINS}
+      CORS_ALLOWED_METHODS: ${COOKIES_REFRESH_MS}
+    depends_on:
+      - db
+    networks:
+      - app-network
+    command: >
+      sh -c "npx typeorm-ts-node-commonjs migration:run -d /app/dist/typeorm.config.js && npm run start:prod"
```
Comment (NegatiV-64) [inline docker-compose.yml:35]: typeorm-ts-node-commonjs can run ts files

**Review by NegatiV-64 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**